src/store/reducers/deploymentsReducer.js

```diff
@@ -0,0 +1,146 @@
+      IoTHubManagerService.getDeployments()
+        .map(toActionCreator(redux.actions.updateDeployments, fromAction))
+        .catch(handleError(fromAction))
+  },
+  /** Create a new deployment */
+  createDeployment: {
+    type: 'DEPLOYMENTS_CREATE',
+    epic: fromAction =>
+      IoTHubManagerService.createDeployment(fromAction.payload)
+        .map(toActionCreator(redux.actions.insertDeployment, fromAction))
+        .catch(handleError(fromAction))
+  },
+  /** Delete deployment */
+  deleteDeployment: {
+    type: 'DEPLOYMENTS_DELETE',
+    epic: fromAction =>
+      IoTHubManagerService.deleteDeployment(fromAction.payload)
+        .map(toActionCreator(redux.actions.deleteDeployment, fromAction))
+        .catch(handleError(fromAction))
+  }
+});
+// ========================= Epics - END
+
+// ========================= Schemas - START
+const deploymentSchema = new schema.Entity('deployments');
+const deploymentListSchema = new schema.Array(deploymentSchema);
+// ========================= Schemas - END
+
+// ========================= Reducers - START
+const initialState = { ...errorPendingInitialState, entities: {} };
+
+const insertDeploymentReducer = (state, { payload, fromAction }) => {
+  const { entities: { deployments }, result } = normalize(payload, deploymentSchema);
+  if (state.entities) {
+    return update(state, {
+      entities: { $merge: deployments },
+      items: { $splice: [[0, 0, result]] },
+      ...setPending(fromAction.type, false)
+    });
+  }
+  return update(state, {
+    entities: { $set: deployments },
+    items: { $set: [result] },
+    ...setPending(fromAction.type, false)
+  });
+};
+
+const deleteDeploymentReducer = (state, { payload, fromAction }) => {
+  const idx = state.items.indexOf(payload);
+  return update(state, {
+    entities: { $unset: [payload] },
+    items: { $splice: [[idx, 1]] },
+    ...setPending(fromAction.type, false)
+  });
+};
+
+const updateDeploymentsReducer = (state, { payload, fromAction }) => {
+  const { entities: { deployments }, result } = normalize(payload, deploymentListSchema);
+  return update(state, {
+    entities: { $set: deployments },
+    items: { $set: result },
+    lastUpdated: { $set: moment() },
+    ...setPending(fromAction.type, false)
+  });
+};
+
+/* Action types that cause a pending flag */
+const fetchableTypes = [
+  epics.actionTypes.fetchDeployments,
+  epics.actionTypes.createDeployment,
+  epics.actionTypes.deleteDeployment
+];
+
+export const redux = createReducerScenario({
+  insertDeployment: { type: 'DEPLOYMENT_INSERT', reducer: insertDeploymentReducer },
+  deleteDeployment: { type: 'DEPLOYMENTS_DELETE', reducer: deleteDeploymentReducer },
+  updateDeployments: { type: 'DEPLOYMENTS_UPDATE', reducer: updateDeploymentsReducer },
+  registerError: { type: 'DEPLOYMENTS_REDUCER_ERROR', reducer: errorReducer },
+  resetPendingAndError: { type: 'DEPLOYMENTS_REDUCER_RESET_ERROR_PENDING', reducer: resetPendingAndErrorReducer },
```

Comment:
This one has REDUCER in the name of the type, just as registerError above it. This seems like its probably correct since it is a reducer wide reset (as pending/error is stored all together). 
However, for devices and packages, the word REDUCER is missing from the type name.

Reply:
updated for devices and packages